native-oci-generative-ai/README.md

```diff
@@ -0,0 +1,56 @@
+
+This example illustrates how to use the Generative AI service provided by Oracle Cloud Infrastructure (OCI). It uses a state-of-the-art, customizable large language model to generate text. 
+
+The [source code](./src/main/java/com/oracle/labs/GenerateTextExample.java) is a self-contained demonstration of using a [GenerativeAiInferenceClient](https://docs.oracle.com/iaas/tools/java/latest/com/oracle/bmc/generativeaiinference/GenerativeAiInferenceClient.html) to generate text, based on a prompt.
+
+The [_pom.xml_](pom.xml) file includes necessary dependents (such as the [OCI Java SDK](https://docs.public.oneportal.content.oci.oraclecloud.com/iaas/Content/API/SDKDocs/javasdk.htm)) and configures the build, including the [Native Maven Plugin](https://graalvm.github.io/native-build-tools/latest/maven-plugin.html) to generate a native executable with GraalVM Native Image.
+
+> For more information, see [Generative AI](https://docs.oracle.com/iaas/Content/generative-ai/home.htm).
+
+## Prerequisites
+* [GraalVM](https://www.graalvm.org/downloads/).
+
+* You must have access to access to Generative AI resources in your tenancy/compartment. (For more information, see [Getting Access to Generative AI](https://docs.oracle.com/iaas/Content/generative-ai/iam-policies.htm).)
+
+## Preparing Your Code
+
+Download or clone the GraalVM demos repository:
+```sh
+git clone https://github.com/graalvm/graalvm-demos
+```
+
+Edit the contents of [_GenerateTextExample.java_](./src/main/java/com/oracle/labs/GenerateTextExample.java) as follows:
+
+1. Update the value of the static variable `CONFIG_PROFILE` with the name of your OCI profile name.
+
+2. Update the value of the static variable `COMPARTMENT_ID` with the OCID of a compartment that has access to Generative AI resources.
+
+Alternatively, you can follow the steps in [Generate Text in Generative AI](https://docs.oracle.com/iaas/Content/generative-ai/use-playground-generate.htm):
+
+* (Optional) Enter your own prompt&mdash;you can edit this in the source code later.
+* (Optional) Set values for the parameters&mdash;you can adjust these in the source code later.
+* When you are happy with your prompt and the output, click **View code**, select **Java** as the programming language, then click **Copy code**, paste the code into the file named _GenerateTextExample.java_ and save it.
+    > Your code should resemble [_GenerateTextExample.java_](./src/main/java/com/oracle/labs/GenerateTextExample.java).
+
+## Running Your Code
+
+Run the following command:
+
+```shell
+mvn install exec:java
+```
+
+This command runs the `main()` method in _GenerateTextExample.java_ and generates a response based on the prompt and parameters of the request.
+
+## Running the Example as Native Executable
+
+1. Create a native executable by running the following command:
+    ```shell
+    mvn clean package -Pnative
+    ```
+
+2. Run the native executable, as follows:
+    ```shell
+    ./target/generative-ai
```

Comment:
Could you maybe add sample output so that users see what's supposed to happen when running the demo (either on HotSpot or in native mode).

Reply:
Done